### ISSUE TITLE
Allowing backslashed separators in ids

### ISSUE THREAD

**Issue by sc0ttkclark** [select2/select2#2170]
### Areas for review
- [ ] Initial setValue doubles up, with original set spaces removed after backslashed separator
- [ ] Selections not removed from the select2 container when you click into select, can select and re-select the same item over and over (non-AJAX)
- [ ] Default tokenizer support for backslashed separator

### Problem description
Example:

```
data: [
    { id: 'Austin\\, TX', 'text': 'Austin, TX' },
    { id: 'Dallas\\, TX', 'text': 'Dallas, TX' },
    { id: 'Ft. Worth\\, TX', 'text': 'Ft. Worth, TX' }
]
```

I'm trying to avoid changing the separator from comma to something else, I am pretty close on this one!

I've got a custom initSelection callback that splits the initial value, and then builds an array to send back to callback().

`jQuery( element.val().split( /[^\\],/ ) ).each( function () {`

The issue I'm hitting right now is that the initial value set back to the input ends up being initialized wrong when select2 loads itself after initSelection:

* Initial DOM value: `Austin\, TX`
* Passed into callback() from custom initSelection: `callback( [ { id: 'Austin\\, TX', text: 'Austin, TX' } ] )`
* After select2 initializes: `Austin\,TX,Austin\, TX`
* Expected: `Austin\, TX`

I'm not sure where to hook in further to allow backslashing separators to be used. I'll continue digging in core select2 to see if I can offer up a patch too.

### Tokenizer support

I was able to get tokenizer to support it by running my own callback for it and using this:

`index = input.search( new RegExp( '[^\\' + separator + ']' + separator ) );`

in place of `index = input.indexOf( separator );`

**Comment by kevin-brown** [comment]
This should be fixed in 4.0.

If it is not fixed, please add a :+1: to the ticket to bring it to our attention. Please also include an example that we can test to verify that it is not actually working.

**Comment by SKYhuangjing** [comment]
data: [
    { id: 'Austin, TX', 'text': 'Austin, TX' },
    { id: 'Dallas, TX', 'text': 'Dallas, TX' },
    { id: 'Ft. Worth, TX', 'text': 'Ft. Worth, TX' }
]
 select2 version is 4.0.2
 I can't select it.
